_examples/oauth2.go

```diff
@@ -0,0 +1,364 @@
+package examples
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/go-oauth2/oauth2/v4/errors"
+	"github.com/go-oauth2/oauth2/v4/generates"
+	"github.com/go-oauth2/oauth2/v4/manage"
+	"github.com/go-oauth2/oauth2/v4/server"
+	"github.com/go-oauth2/oauth2/v4/store"
+	"github.com/go-session/session"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+var (
+	clientStore   = store.NewClientStore()
+	authConfig    = new(oauth2.Config)
+	globalToken   *oauth2.Token
+	authServerURL string
+)
+
+const (
+	usernameConfig = "test"
+	passwordConfig = "test"
+	userIDConfig   = "123456"
+)
+
+// AuthServerHandler is a simple http.Handler that implements go-oauth2 server.
+func AuthServerHandler() http.Handler {
+	manager := manage.NewDefaultManager()
+	manager.SetAuthorizeCodeTokenCfg(manage.DefaultAuthorizeCodeTokenCfg)
+
+	manager.MustTokenStorage(store.NewMemoryTokenStore())
+
+	manager.MapAccessGenerate(generates.NewAccessGenerate())
+
+	manager.MapClientStorage(clientStore)
+
+	srv := server.NewServer(server.NewConfig(), manager)
+
+	srv.SetPasswordAuthorizationHandler(
+		func(ctx context.Context,
+			clientID,
+			username,
+			password string) (userID string, err error) {
+			if username == usernameConfig && password == passwordConfig {
+				userID = userIDConfig
+			}
+			return
+		},
+	)
```

Comment:
Since other handlers are extracted into functions, maybe extract this one too?

Reply:
Addressed